### PR TITLE
Logger 2.3 refactor

### PR DESCRIPTION
**Pull request by luke-hill**
- Bump Minimum Ruby to 2.3
- Use Ruby 2.3 logger implementation (Much better)
- Rubocop / refactoring done according to new min ruby version

### CODE REVIEW

**Comment by luke-hill** [comment]
Ping @tgaff / @ineverov - Bout time we had a new feature and a chunky change to this gem. So here it is.

Note this forces the minimum ruby up 1 point again. I'll make additional changes in other PR's to tweak docs / travis versions accordingly.

File: features/step_definitions/implicit_waiting_steps.rb [hunk 1]
```diff
@@ -11,21 +9,21 @@
...
-  @test_site.slow.first_section(count: 1)
+  @test_site.slow.first_section
 
   expect(Time.now - start_time).to be_between(0.15, 0.3)
 end
 
 Then('the slow sections are waited for') do
   start_time = Time.now
-  @test_site.slow.all_sections(count: 2)
+  @test_site.slow.all_sections
```
Comment (ineverov) [inline features/step_definitions/implicit_waiting_steps.rb]: should we check count here?

File: features/step_definitions/implicit_waiting_steps.rb [hunk 2]
```diff
@@ -49,15 +47,15 @@
...
+  expect(@test_site.slow.has_first_section?).to be true
 
   expect(Time.now - start_time).to be_between(0.15, 0.3)
 end
 
 Then('the boolean test for slow sections are waited for') do
   start_time = Time.now
 
-  expect(@test_site.slow.has_all_sections?(count: 2)).to be true
+  expect(@test_site.slow.has_all_sections?).to be true
```
Comment (ineverov) [inline features/step_definitions/implicit_waiting_steps.rb]: `count: 2` here?

File: lib/site_prism.rb [hunk 3]
```diff
@@ -13,18 +13,55 @@ module SitePrism
...
-      @enable_logging ||= false
+    # `Logger#reopen` was added in Ruby 2.3 - Which is now the minimum version
+    # for the site_prism gem
+    #
+    # This writer method allows you to configure where you want the output of
+    # the site_prism logs to go (Default is $stdout)
+    #
+    # example: SitePrism.log_output = 'site_prism.log' would save all
+    # log messages to `./site_prism.log`
+    def log_output=(logdev)
```
Comment (ineverov) [inline lib/site_prism.rb]: Maybe better name is `log_path`?
Or is there any specific use case?

File: site_prism.gemspec [hunk 4]
```diff
@@ -5,7 +5,7 @@ require './lib/site_prism/version'
 Gem::Specification.new do |s|
   s.name        = 'site_prism'
   s.version     = SitePrism::VERSION
-  s.required_ruby_version = '>= 2.2'
+  s.required_ruby_version = '>= 2.3'
```
Comment (ineverov) [inline site_prism.gemspec]: align

File: features/step_definitions/implicit_waiting_steps.rb (hunk 1 above)
Comment (luke-hill) [reply features/step_definitions/implicit_waiting_steps.rb]: I suppose there's no harm in re-adding this one back. It was just we needed these in to make it pass on Capybara 2.x

I'll re-add it

File: lib/site_prism/section.rb [hunk 5]
```diff
@@ -36,7 +36,10 @@ def initialize(parent, root_element)
     # occurs and calls within a section (For example section.find(element))
     # correctly scope to look within the section only
     def page
-      root_element || super
+      return root_element if root_element
+
+      SitePrism.logger.warn('Root Element not found; Falling back to `super`')
```
[the remark this answers is not in the document]
Comment (luke-hill) [reply lib/site_prism/section.rb]: Just going off the Logger defaults here, probably gonna keep this as is.

File: site_prism.gemspec (hunk 4 above)
Comment (luke-hill) [reply site_prism.gemspec]: To get this file all aligned correctly is one that i'll let someone else tackle (If you wanna tackle it go for it), as there's a few different ways you can do it.

**Comment by luke-hill** [comment]
Re-review @ineverov now I've fixed everything up. CI is green and made review points

**Pull request merged**